x-pack/plugins/apm/public/utils/url.js

```diff
@@ -16,19 +16,34 @@ import { EuiLink } from '@elastic/eui';
 import createHistory from 'history/createHashHistory';
 import chrome from 'ui/chrome';
 
-export function getMlJobUrl(serviceName, transactionType, location) {
+export function ViewMLJob({
+  serviceName,
+  transactionType,
+  location,
+  children = 'View Job'
+}) {
   const { _g, _a } = decodeKibanaSearchParams(location.search);
-  const nextSearch = encodeKibanaSearchParams({
+
+  const pathname = '/app/ml';
+  const hash = '/timeseriesexplorer';
+  const query = {
     _g: {
       ..._g,
       ml: {
         jobIds: [`${serviceName}-${transactionType}-high_mean_response_time`]
       }
     },
     _a
-  });
+  };
 
-  return `/app/ml#/timeseriesexplorer/?${nextSearch}`;
+  return (
+    <KibanaLink
+      pathname={pathname}
+      hash={hash}
+      query={query}
+      children={children}
+    />
+  );
 }
```

Comment:
@jasonrhodes At least now it is based on the `KibanaLink` component. Is that ok with you?

Reply:
Totally fine with me -- sorry I missed this not working in the other PR!

Reply:
No worries. I should have manually tested it after making the change.